src/storage/datasets_index.rs

```diff
@@ -43,206 +44,62 @@ impl DatasetsIndex {
+                        panic!("Couldn't parse header name: {}: {e:?}", k)
+                    }),
+                    reqwest::header::HeaderValue::from_str(&v).unwrap_or_else(|e| {
+                        panic!("Couldn't parse header value: {}: {e:?}", v)
+                    }),
+                )
+            }).collect() 
+        }
+    }
 
-    #[test]
-    fn test_assignment_to_chunk_set() {
-        let dataset_1 = "s3://ethereum-mainnet".to_string();
-        let dataset_2 = "s3://moonbeam-evm".to_string();
-        let assignment = WorkerAssignment {
-            dataset_chunks: vec![
-                DatasetChunks {
-                    dataset_id: dataset_1.clone(),
-                    chunks: vec![
-                        AssignedChunk {
-                            path: "0000000000/0000000000-0000697499-f6275b81".to_string(),
-                            filenames: vec![0, 1, 2],
-                        },
-                        AssignedChunk {
-                            path: "0000000000/0000697500-0000962739-4eff9837".to_string(),
-                            filenames: vec![0, 1, 2],
-                        },
-                        AssignedChunk {
-                            path: "0007293060/0007300860-0007308199-6aeb1a56".to_string(),
-                            filenames: vec![0, 1, 2],
-                        },
-                    ],
-                    download_url: "https://example.com".to_string(),
-                    ..Default::default()
-                },
-                DatasetChunks {
-                    dataset_id: dataset_2.clone(),
-                    chunks: vec![
-                        AssignedChunk {
-                            path: "0000000000/0000190280-0000192679-0999c6ce".to_string(),
-                            filenames: vec![0, 1, 2, 3, 4],
-                        },
-                        AssignedChunk {
-                            path: "0003510340/0003515280-0003518379-b8613f00".to_string(),
-                            filenames: vec![0, 1, 2, 3, 4],
-                        },
-                    ],
-                    download_url: "https://example.com".to_string(),
-                    ..Default::default()
-                },
-            ],
-            http_headers: vec![HttpHeader {
-                name: "Auth".to_string(),
-                value: "None".to_string(),
-            }],
-            known_filenames: vec![
-                "blocks.parquet".to_string(),
-                "transactions.parquet".to_string(),
-                "logs.parquet".to_string(),
-                "statediffs.parquet".to_string(),
-                "traces.parquet".to_string(),
-            ],
+    pub fn create_chunks_set(&self) -> ChunkSet {
+        let mut chunk_set = ChunkSet::new();
+        for (dataset_id, dataset_index) in &self.datasets {
+            for data_chunk in dataset_index.files.keys() {
```

Comment:
The name `files` is a bit confusing. Can we name it `chunks` or `files_by_chunk`?

Reply:
done